libfprint/drivers/goodixtls/5395/goodix_device.c

```diff
@@ -105,34 +109,39 @@ static gboolean fpi_goodix_device_receive_chunk(FpDevice *dev, guint8 **data, gl
     }
 
     if (success) {
-        memcpy(*data, transfer->buffer, transfer->actual_length);
-        *length = transfer->actual_length;
-        fp_dbg("Received chunk %s", fpi_goodix_protocol_data_to_str(*data, *length));
+        g_byte_array_append(data, transfer->buffer, transfer->actual_length);
+        fp_dbg("Received chunk %s", fpi_goodix_protocol_data_to_str(transfer->buffer, transfer->actual_length));
     }
     fpi_usb_transfer_unref(transfer);
     return success;
 }
 
 gboolean fpi_goodix_device_receive_data(FpDevice *dev, GoodixMessage **message, GError **error) {
-    guint8 *data = g_malloc0( GOODIX_EP_IN_MAX_BUF_SIZE);
-    glong length = 0;
+    GByteArray *buffer = g_byte_array_new();
     glong message_length = 0;
-    guint chunk_count = 1;
-
-    if (fpi_goodix_device_receive_chunk(dev, &data, &length, error)) {
-        GoodixDevicePack *pack = (GoodixDevicePack *) data;
+    guint8 command_byte;
+    if (fpi_goodix_device_receive_chunk(dev, buffer, error)) {
+        GoodixDevicePack *pack = (GoodixDevicePack *) buffer->data;
+        command_byte = pack->cmd;
         message_length = pack->length;
     } else {
         return FALSE;
     }
 
-    while (length - 1 < message_length) {
-        chunk_count++;
-        data = g_realloc(data, GOODIX_EP_IN_MAX_BUF_SIZE * chunk_count);
-        fpi_goodix_device_receive_chunk(dev, (&data + (chunk_count - 1) * GOODIX_EP_IN_MAX_BUF_SIZE), &length, error);
+    while (buffer->len - 1 < message_length) {
+        GByteArray *chuck = g_byte_array_new();
```

Comment:
This should be `chunk`, right?

Reply:
No, g_byte_array_new() returns a pointer to a certain size of the heap, I think that this should be the correct type

Reply:
I think this is typo in variable name 'chuck' -> should be 'chunk'

Reply:
Ah of course, sorry :)